systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/KafkaRollerST.java

```diff
@@ -322,6 +332,143 @@ void testKafkaPodPendingDueToRack(ExtensionContext extensionContext) {
+     *  3. - Take snapshots of the broker and controller pods for later comparison.
+     *  4. - Update a specific Kafka configuration that affects only controller nodes and verify the rolling update behavior.
+     *     - Ensure that only controller nodes undergo a rolling update, while broker nodes remain unaffected.
+     *  5. - Introduce a change in the controller node pool, such as modifying pod affinity.
+     *     - Observe and ensure that this change triggers another rolling update for the controller nodes.
+     *  6. - Verify the rolling updates of controller nodes by comparing the snapshots taken before and after each configuration change.
+     *
+     * @usecase
+     *  - kafka-controller-node-rolling-update
+     *  - kafka-configuration-change-impact
+     *  - kafka-node-pool-property-update
+     *  - kafka-node-pool-management
+     */
+    @ParallelNamespaceTest
+    void testKafkaRollingUpdatesOfControllerNodes(final ExtensionContext extensionContext) {
+        assumeTrue(Environment.isKRaftModeEnabled());
+        assumeFalse(Environment.isOlmInstall() || Environment.isHelmInstall());
+
+        final TestStorage testStorage = new TestStorage(extensionContext);
+        final String brokerPoolName = testStorage.getKafkaNodePoolName() + "-b";
+        final String controllerPoolName = testStorage.getKafkaNodePoolName() + "-c";
+
+        final int brokerPoolReplicas = 3, controllerPoolReplicas = 3;
+
+        final Kafka kafka = KafkaTemplates.kafkaPersistent(testStorage.getClusterName(), 1, 1).build();
+        final KafkaNodePool brokerPool = KafkaNodePoolTemplates.kafkaBasedNodePoolWithBrokerRole(brokerPoolName, kafka, brokerPoolReplicas).build();
+        final KafkaNodePool controllerPool = KafkaNodePoolTemplates.kafkaBasedNodePoolWithControllerRole(controllerPoolName, kafka, controllerPoolReplicas).build();
+
+        final LabelSelector brokerPoolSelector = KafkaNodePoolResource.getLabelSelector(testStorage.getClusterName(), brokerPoolName, ProcessRoles.BROKER);
+        final LabelSelector controllerPoolSelector = KafkaNodePoolResource.getLabelSelector(testStorage.getClusterName(), controllerPoolName, ProcessRoles.CONTROLLER);
+
+        resourceManager.createResourceWithoutWait(extensionContext, brokerPool, controllerPool, kafka);
+
+        PodUtils.waitForPodsReady(testStorage.getNamespaceName(), brokerPoolSelector, brokerPoolReplicas, true);
+        PodUtils.waitForPodsReady(testStorage.getNamespaceName(), controllerPoolSelector, controllerPoolReplicas, true);
+
+        Map<String, String> brokerPoolPodsSnapshot = PodUtils.podSnapshot(testStorage.getNamespaceName(), brokerPoolSelector);
+        Map<String, String> controllerPoolPodsSnapshot = PodUtils.podSnapshot(testStorage.getNamespaceName(), controllerPoolSelector);
+
+        // change Controller-only configuration inside shared Kafka configuration between KafkaNodePools and see that only mixed and controller pods rolls
```

Comment:
```suggestion
        // change Controller-only configuration inside shared Kafka configuration between KafkaNodePools and see that only controller pods rolls
```

Reply:
also just curious, do we not want to make config change that only applies to broker only pods as well and verify that controller only pods do not roll?